tests/integration/schema_registry/test_api.py

```diff
@@ -0,0 +1,257 @@
+
+    Raises:
+        AssertionError if the returned schema and id do not match the
+        registration schema and schema id.
+
+    """
+    sr_conf = SchemaRegistryConfig(kafka_cluster.schema_registry())
+    sr = SchemaRegistryClient(sr_conf)
+
+    schema = load_avsc('basic_schema.avsc')
+    subject = _subject_name('test_registration')
+
+    version = sr.register_schema(subject, schema)
+    version2 = sr.get_registration(subject, version.schema)
+
+    assert version.schema == version2.schema
+    assert version2.subject == version2.subject
+    assert version.version == version2.version
+
+
+def test_api_delete_subject_version(kafka_cluster, load_avsc):
+    """
+    Registers a Schema under a specific subject then deletes it.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the version number is not returned in the response or
+        if the subject has not been deleted.
+
+    """
+    sr_conf = SchemaRegistryConfig(kafka_cluster.schema_registry())
+    sr = SchemaRegistryClient(sr_conf)
+
+    schema = load_avsc('basic_schema.avsc')
+    subject = str(uuid1())
+
+    sr.register_schema(subject, schema)
+    sr.delete_version(subject, 1)
+
+    assert subject not in sr.get_subjects()
+
+
+def test_api_subject_config_update(kafka_cluster, load_avsc):
+    """
+    Updates a subjects compatibility policy then ensures the same policy
+    is returned when queried.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the compatibility policy returned by the schema
+        registry fixture does not match the updated value.
+
+    """
+    sr_conf = SchemaRegistryConfig(kafka_cluster.schema_registry())
+    sr = SchemaRegistryClient(sr_conf)
+
+    schema = load_avsc('basic_schema.avsc')
+    subject = str(uuid1())
+
+    sr.register_schema(subject, schema)
+    sr.set_compatibility(subject_name=subject, level=CompatibilityType.FULL_TRANSITIVE)
+
+    assert sr.get_compatibility(subject_name=subject) == CompatibilityType.FULL_TRANSITIVE
+
+
+def test_api_config_update(kafka_cluster, load_avsc):
+    """
+    Updates a global compatibility policy then ensures the same policy
+    is returned when queried.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the compatibility policy returned by the schema
+        registry fixture does not match the updated value.
+
+    """
+    sr_conf = SchemaRegistryConfig(kafka_cluster.schema_registry())
+    sr = SchemaRegistryClient(sr_conf)
+
+    sr.set_compatibility(level=CompatibilityType.FORWARD_TRANSITIVE)
+
+    assert sr.get_compatibility() == CompatibilityType.FORWARD_TRANSITIVE
```

Comment:
There are no negative tests, i.e., that's that raise errors: bad SR URL, bad SR config, invalid request, etc.